pvlib/modelchain.py

```diff
@@ -366,14 +396,37 @@ def __init__(self, system, location,
 
         self.weather = None
         self.times = None
-        self.solar_position = None
+
+        self.results = ModelChainResult()
 
         if kwargs:
             warnings.warn(
                 'Arbitrary ModelChain kwargs are deprecated and will be '
                 'removed in v0.9', pvlibDeprecationWarning
             )
 
+    def __getattr__(self, key):
+        if key in ModelChain._deprecated_attrs:
+            msg = f'ModelChain.{key} is deprecated and will' \
+                  f' be removed in v0.10. Use' \
+                  f' ModelChain.results.{key} instead'
+            warnings.warn(msg, pvlibDeprecationWarning)
+            return getattr(self.results, key)
+        else:
+            try:
+                return self.__dict__[key]
+            except(KeyError):
+                raise AttributeError
```

Comment:
I actually think the entire `try/except` block is not necessary.  Since `__getattr__` is only called when the "normal" `__getattribute__` method raises an `AttributeError` (see [docs](https://docs.python.org/3/reference/datamodel.html#object.__getattr__)) we can just raise an attribute error if the requested attribute is not part of `ModelChainResults`.

I would appreciate someone else's thoughts on this though, in case I'm overlooking some nuance of the instance dictionary.